token/selector.go

```diff
@@ -25,21 +26,31 @@ var (
 	SelectorSufficientFundsButConcurrencyIssue = errors.New("sufficient funds but concurrency issue")
 )
 
-// OwnerFilter tells if a passed identity is recognized
-type OwnerFilter interface {
+// SelectorFilter is the base interface for a token selector filter
+type SelectorFilter interface{}
+
+// SelectorFilterByID is a token selector filter that selects tokens by owner ID
+type SelectorFilterByID interface {
+	SelectorFilter
+	// ID returns the ID of the identity the selection should select for.
 	ID() string
+}
+
+// SelectorFilterByUnspentToken is a SelectorFilter that selects tokens that are not yet spent.
+type SelectorFilterByUnspentToken interface {
+	SelectorFilter
 	// ContainsToken returns true if the passed token is recognized, false otherwise.
 	ContainsToken(token *token2.UnspentToken) bool
 }
```

Comment:
Mhhh. The filter used here is only there to dispatch the select call to either to selectByOwnerID or selectByFilter.

My understanding of a filter is something that once applies on a data structure and there is a predicate to be true in order to get the item or not. this is not the case with the selectorFilter. Also, the filter may have more predicates. For instance, tokenType could be one. We maybe wanna add more filters, like minTokenSize, maxTokenSize, oldestToken, …. In order to model is nicely, we could throw away the filter interface it is right now and make it just a function, or a list of functions.  WDYT?

Reply:
@adecaro is the ID of the identity always a string? In some code I saw that this requires to do something some casting of the actual identity using the "raw material" `string(Owner.Raw)`.